docs/specification/draft/server/profiles.mdx

```diff
@@ -0,0 +1,137 @@
+
+```
+/.well-known/mcp-profiles
+```
+
+betewen the "authority" component (i.e. the domain name or IP address and port) and any path component of the MCP Server's URL.
+
+So for example, if your MCP server is located at: `https://mymcpserver.mycompany.com`, then the profiles declaration document is at: `https://mymcpserver.mycompany.com/.well-known/mcp-profiles`. If your MCP server is located at: `https://mycompany.com/mymcpserver`, then the profiles declaration document is at: `https://mycompany.com/.well-known/mcp-profiles/mymcpserver`.
+
+### Backward compatibility
+If this resource does not exist on the MCP Server, or if the document returned is empty, then the Client MUST assume that the MCP Server does not support Profiles.
+
+## Profile Discovery
+Before connecting to an MCP Server, an MCP Client MAY fetch the Server’s Well Known URI to understand which profiles the server supports (if any).
+
+## Initialization With Profiles
+The MCP Client makes an `InitializeRequest` based on its knowledge of which (if any) profiles the MCP Server supports.
+
+### Requested Profiles
+
+The `InitializeRequest` object has an optional `requestedProfiles` field, whose value is an array of profile URLs.
+
+```TypeScript
+interface InitializeRequest extends Request {
+  method: "initialize";
+  params: {
+    protocolVersion: string;
+    capabilities: ClientCapabilities;
+    clientInfo: Implementation;
+    requestedProfiles?: [string];  // Optional field to request specific profiles.
+  };
+}
+```
+
+If the Client has no opinion of which profile to use, it omits the `requestedProfiles` field from the `InitializeRequest`. Otherwise, the Client specifies one or more profiles in the value `requestedProfiles` field of the `InitializeRequest` it sends to the MCP Server. The order of the profiles in the array is the Client's order of preference of profiles.
+
+### Selected Profile
+
+The `InitializeResult` interface has an optional `profile` field:
+
+```TypeScript
+interface InitializeResult extends Result {
+  protocolVersion: string;
+  capabilities: ServerCapabilities;
+  serverInfo: Implementation;
+  instructions?: string;
+  profile?: string;  // When present, this field specifies the selected profile.
+}
+```
+
+The value of the `profile` field is the Profile URL of the selected Profile. If this field is omitted in the response, and the MCP Server has a Profiles Declaration, then the Client MUST assume that the MCP Server has chosen the default profile specified in its Profiles Declaration. If the MCP Server does not have a Profiles Declaration, then the `InitializeResult` MUST NOT have a `profile` field.
+
+## Profile Negotiation
+
+1. The Client discovers the profiles supported by an MCP Server, optionally by performing a `HTTP GET` on the MCP Server's well-known Profiles Declaration URL.
```

Comment:
You're saying that by optionally doing a HTTP `GET`. What are the alternatives?

Reply:
Added a line to clarify that an MCP Client might already know which profiles a server supports, and therefore not go through the discovery process.